src/videojs-resumecard.js

```diff
@@ -23,43 +27,38 @@ let resumeCard = null;
- * @param {string} [options.classNameActionItem=vjs-resume-card-action-item] HTMLElement class
- * @param {string} [options.classNameActionList=vjs-resume-card-action-list] HTMLElement class
- * @param {string} [options.classNameButton=vjs-resume-card-button] HTMLElement class
- * @param {string} [options.classNameRestartButton=vjs-resume-card-restart-button] HTMLElement class
- * @param {string} [options.classNameResumeButton=vjs-resume-card-resume-button] HTMLElement class
- * @param {function(restartCallback):HTMLElement} [options.getRestartButton] Callback to return custom restart button HTMLElement.
- * @param {function(resumeCallback):HTMLElement} [options.getResumeButton] Callback to return custom resume button HTMLElement.
- * @param {function(restartButton:HTMLElement, resumeButton:HTMLElement):HTMLElement} [options.getTemplate] Callback to return custom template HTMLElement.
- * @param {string} [options.id=vjs_resume_card] HTMLElement id
- * @param {string} [options.restartButtonText=Restart Video]
+ * @param {string} [options.className=vjs-resumecard] HTMLElement class.
+ * @param {string} [options.id=vjs_resumecard] HTMLElement id.
+ * @param {boolean} [options.preload] Have the player *immediately* seek to time specified.
  * @param {function(event)} [options.restartCallback] Custom restart button callback.
- * @param {string} [options.resumeButtonText=Resume Video]
  * @param {function(event)} [options.resumeCallback] Custom resume button callback.
+ * @param {function(restartCallback, resumeCallback):HTMLElement} options.template Callback to return custom template HTMLElement.
  * @param {number} options.time Start time, in seconds, for the contented to be resumed.
  */
 class ResumeCard {
   constructor(player, options = {}) {
     if (!options.time) {
-      throw new Error('videojs-resumecard requires a resume time.');
+      throw new Error('videojs-resumecard requires a valid time.');
+    }
+
+    if (!options.template) {
+      throw new Error('videojs-resumecard requires a valid template.');
     }
 
+    // hide big play button since resume card will take over initial play experience
     if (player.bigPlayButton) {
       player.bigPlayButton.hide();
     }
 
+    // seek immediately on preload
+    if (options.preload === true) {
+      player.currentTime(options.time);
+    }
+
     this.player = player;
     this.settings = videojs.mergeOptions({
-      className: 'vjs-resume-card',
-      classNameActionItem: 'vjs-resume-card-action-item',
-      classNameActionList: 'vjs-resume-card-action-list',
-      classNameButton: 'vjs-resume-card-button',
-      classNameRestartButton: 'vjs-resume-card-restart-button',
-      classNameResumeButton: 'vjs-resume-card-resume-button',
-      id: 'vjs_resume_card',
-      restartButtonText: 'Restart Video',
-      resumeButtonText: 'Resume Video',
+      className: 'vjs-resumecard',
+      id: 'vjs_resumecard',
```

Comment:
Maybe outside the scope of this PR, but I don't think we should be adding an `id` to this component if we can help it.  Since `id`s must be unique on the page, if someone wanted to use this on a player that had multiple instances on the same page, it would be cumbersome.